host/host.go

```diff
@@ -0,0 +1,38 @@
+package host
+
+import (
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/taskcluster/shell"
+)
+
+func RunCommand(cmd *exec.Cmd) (combinedOutput string, err error) {
+	log.Print("Running command: " + shell.Escape(cmd.Args...))
+	out, err := cmd.CombinedOutput()
```

Comment:
We're not actually running `shell.Escape()` on the `cmd.Args` that gets run here, just in the log printout - it seems like we should escape the actual commandline, right?

Reply:
The `shell.Escape` function just outputs command arguments in a shell escaped fashion, such that you could copy/paste the output into a bash shell, and the arguments would be escaped correctly. `cmd.Args` is the component of the command that contains the arguments (including the program itself), as opposed to e.g. the env vars or other state associated with the command.

Reply:
I think ideally we prevent the kinds of shell escapes that can allow user submitted payloads to do scary things on workers. Something like `shell.Escape()` can be used to do that.

These kinds of scary things: https://github.com/taskcluster/generic-worker/blob/master/process/docker.go#L62-L63